VVResonances/interactive/runFitPlots.py

```diff
@@ -163,84 +253,39 @@ def doZprojection(pdfs,data,norm,zBinslowedge,xBins_redux,yBins_redux,zBins_redu
              MJ2.setVal(yv)
              for zk,zv in zBins_redux.iteritems():
                  MJJ.setVal(zv)
-                 dh.Fill(zv,data.weight(argset))
+                 #dh.Fill(zv,data.weight(argset))
+		 neventsPerBin[zk-1] += data.weight(argset)
                  i=0
                  binV = zBinsWidth[zk]*xBinsWidth[xk]*yBinsWidth[yk]
                  for p in pdfs:
-                    if "pdfdata" in p.GetName():
-                            lv[i][zv] += p.weight(argset)
-                    else:
-                            lv[i][zv] += p.getVal(argset)*binV
+                    lv[i][zv] += p.getVal(argset)*binV
                     i+=1
     for i in range(0,len(pdfs)):
         for zk,zv in zBins_redux.iteritems():
-            if "pdfdata" in pdfs[i].GetName():
-                h[i].Fill(zv,lv[i][zv])
-            else:
-                h[i].Fill(zv,lv[i][zv]*norm)
-    leg = ROOT.TLegend(0.88,0.65,0.7,0.88)
-    c = ROOT.TCanvas("c","c",800,400)
-    if postfit:
-        pad1 = ROOT.TPad("pad1", "pad1", 0, 0.3, 1, 1.0)
-        pad1.SetBottomMargin(0.01)
-        pad1.SetLogy()
-        pad1.Draw()
-        pad1.cd()    
-    c.SetLogy()
-    h[0].SetLineColor(colors[0])
-    h[0].SetTitle("Z-Proj. x : "+options.xrange+" y : "+options.yrange)
-    h[0].GetXaxis().SetTitle("m_{jj}")
-    h[0].GetYaxis().SetTitleOffset(1.3)
-    h[0].GetYaxis().SetTitle("events")
-    if postfit:
-        h[0].GetYaxis().SetTitleOffset(0.6)
-        h[0].GetYaxis().SetTitle("events")
-        h[0].GetYaxis().SetTitleSize(0.06)
-        h[0].GetYaxis().SetLabelSize(0.06)
-        h[0].GetYaxis().SetNdivisions(5)
-    h[0].Draw("hist")
+           if "nonRes" in str(pdfs[i].GetName()):
```

Comment:
I would suggest adding a few more posibilties here otherwise some of the pdfs from the workspace won't have the right normalizations: 
if "nonRes" in str(pdfs[i].GetName()):
	     print "nonRes: ",pdfs[i].GetName()
	     h[i].Fill(zv,lv[i][zv]*norm_nonres)
	   if "VJets" in str(pdfs[i].GetName()):
	     print "Res: ",pdfs[i].GetName()
	     h[i].Fill(zv,lv[i][zv]*norm_res)
           if "model_b" in str(pdfs[i].GetName()):
	     print "all bkg contributions: ",pdfs[i].GetName()
	     h[i].Fill(zv,lv[i][zv]*(norm_res+norm_nonres))
           if "model_s" in str(pdfs[i].GetName()):
	     print "normalize full model: ",pdfs[i].GetName()
	     h[i].Fill(zv,lv[i][zv]*(norm_res+norm_nonres+norm_s)) 
at least model_b and model_s would be important to normalize correctly (at least for me since i use this script to also plot stuff for the toys) of course that also means adding the signal normalization to each of the functions

Reply:
The script as it is now does not consider model_s while it only fit and plots model_b. But please, go ahead in adding the signal component. Perhaps a flag could be added to choose b-only fit or s+b fit (or both) ?